sklearn/linear_model/tests/test_logistic.py

```diff
@@ -542,12 +542,12 @@ def test_logistic_regressioncv_class_weights():
     X, y = make_classification(n_samples=20, n_features=20, n_informative=10,
                                n_classes=3, random_state=0)
 
-    # Test the liblinear fails when class_weight of type dict is
-    # provided, when it is multiclass. However it can handle
-    # binary problems.
+    msg = ("In LogisticRegressionCV the liblinear solver cannot handle "
```

Comment:
here is the problem : 
- `_fit_liblinear(loss='logistic_regression')` does not handle `sample_weight`, but handles `class_weight` directly, and only with binary problems (!) .

- SAG, LBFGS and Newton-cg solvers handles `class_weight` by putting the weights into the `sample_weight`.

That is why we have a different handling of `class_weight` for liblinear.
___
Then, there are 3 cases, with `solver='liblinear'`:
 1. **multinomial case**, but it raise an error in `_check_solver_option`.
 2. **binary case**, then the `class_weight` dictionnary is reconstructed with keys -1 and 1, probably in order to match the API of `_fit_liblinear`
 3. **OvR with `n_classes > 2` case**, and then it raises an error (@MechCoder). The reason is that the other solvers use the class weight for each sample, but it is not possible in `fit_liblinear` since it does not handles `sample_weight` yet. With the 'balanced' case, the class weights are computed **after** the binarization one-vs-rest, which can be used in the same way by all solvers, liblinear included.

___
**Conclusion:** We probably just need to merge #5274 to be consistent with all solvers in `LogisticRegression`

The inconsistency for `class_weight='balanced'` in the OvR case (inconsistency between SGDClassifier, LinearSVC and LogisticRegression) is another topic (see https://github.com/scikit-learn/scikit-learn/pull/5008#issuecomment-139288413)